pybess/bess.py

```diff
@@ -192,20 +199,24 @@ def connect(self, grpc_url=DEF_GRPC_URL):
         if self.is_connected():
             raise self.APIError('Already connected')
 
-        self.status = None
-        self.peer = grpc_url
-        self.channel = grpc.insecure_channel(grpc_url)
-        self.channel.subscribe(self._update_status, try_to_connect=True)
-        self.stub = service_pb2.BESSControlStub(self.channel)
-
         while not self.is_connected():
-            if self.status in [grpc.ChannelConnectivity.TRANSIENT_FAILURE,
-                               grpc.ChannelConnectivity.SHUTDOWN,
-                               self.BROKEN_CHANNEL]:
-                self.disconnect()
-                raise self.APIError(
-                    'Connection to {} failed'.format(grpc_url))
-            time.sleep(0.1)
+            self.status = None
+            self.peer = grpc_url
+            self.channel = grpc.insecure_channel(grpc_url)
+            self.channel.subscribe(self._update_status, try_to_connect=True)
+            self.stub = service_pb2.BESSControlStub(self.channel)
+
+            while not self.is_connected():
```

Comment:
The logic with these two nested `while not self.is_connected()` seems a bit tortured. Would it be possible to restructure this as:

    while not self.is_connected():
        if <some condition>:
            do the status, peer, channel, stub setup
        if <other conditions>: # or `elif`?
            disconnect, maybe raise error
        time.sleep(0.1)

?

Reply:
Maybe changing the first `while not` can be replaced with `if` also good.

Reply:
Thanks, revised. The `while` cannot be replaced with `if` in case of involuntary disconnection.